From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: acrylic-style <me@acrylicstyle.xyz>
Date: Mon, 11 Jan 2021 03:32:09 +0900
Subject: [PATCH] magmacube > vanilla!


diff --git a/src/main/java/net/minecraft/DetectedVersion.java b/src/main/java/net/minecraft/DetectedVersion.java
index 364b02952a3b85e8356772dfe9e9aaefadc9d217..26b2e932049ccbacd747c8eccb21f75ddc7983a0 100644
--- a/src/main/java/net/minecraft/DetectedVersion.java
+++ b/src/main/java/net/minecraft/DetectedVersion.java
@@ -25,6 +25,7 @@ public class DetectedVersion implements WorldVersion {
     private final int resourcePackVersion;
     private final int dataPackVersion;
     private final Date buildTime;
+    @javax.annotation.Nullable public final String magmaCubeVersion; // MagmaCube
 
     private DetectedVersion() {
         this.id = UUID.randomUUID().toString().replaceAll("-", "");
@@ -35,6 +36,7 @@ public class DetectedVersion implements WorldVersion {
         this.resourcePackVersion = 18;
         this.dataPackVersion = 18;
         this.buildTime = new Date();
+        this.magmaCubeVersion = "debug"; // MagmaCube
     }
 
     private DetectedVersion(JsonObject json) {
@@ -47,6 +49,7 @@ public class DetectedVersion implements WorldVersion {
         this.resourcePackVersion = GsonHelper.getAsInt(jsonObject2, "resource");
         this.dataPackVersion = GsonHelper.getAsInt(jsonObject2, "data");
         this.buildTime = Date.from(ZonedDateTime.parse(GsonHelper.getAsString(json, "build_time")).toInstant());
+        this.magmaCubeVersion = json.has("magmacube_version") ? GsonHelper.getAsString(json, "magmacube_version") : "debug"; // MagmaCube
     }
 
     public static WorldVersion tryDetectVersion() {
diff --git a/src/main/java/net/minecraft/SystemReport.java b/src/main/java/net/minecraft/SystemReport.java
index a506f452e9357dd504701050482d3495e80fb8df..d407afcacaad89de4836abdc8a975f73570614ef 100644
--- a/src/main/java/net/minecraft/SystemReport.java
+++ b/src/main/java/net/minecraft/SystemReport.java
@@ -27,6 +27,7 @@ public class SystemReport {
     private final Map<String, String> entries = Maps.newLinkedHashMap();
 
     public SystemReport() {
+        this.setDetail("MagmaCube Version", ((DetectedVersion) SharedConstants.getCurrentVersion()).magmaCubeVersion); // MagmaCube
         this.setDetail("Minecraft Version", SharedConstants.getCurrentVersion().getName());
         this.setDetail("Minecraft Version ID", SharedConstants.getCurrentVersion().getId());
         this.setDetail("Operating System", OPERATING_SYSTEM);
diff --git a/src/main/java/net/minecraft/client/ClientBrandRetriever.java b/src/main/java/net/minecraft/client/ClientBrandRetriever.java
index a71fa408ba7e733c02f450dd5c18a17bee53a427..fb84bbcd532981037e97b33e7b8bdef539dbbd2c 100644
--- a/src/main/java/net/minecraft/client/ClientBrandRetriever.java
+++ b/src/main/java/net/minecraft/client/ClientBrandRetriever.java
@@ -3,10 +3,10 @@ package net.minecraft.client;
 import net.minecraft.obfuscate.DontObfuscate;
 
 public class ClientBrandRetriever {
-    public static final String VANILLA_NAME = "vanilla";
+    public static final String VANILLA_NAME = "magmacube"; // MagmaCube > vanilla!
 
     @DontObfuscate
     public static String getClientModName() {
-        return "vanilla";
+        return VANILLA_NAME;
     }
 }
\ No newline at end of file
diff --git a/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java b/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
index 2fb422f45c4eab5f7a4912c0c5eaae9292565438..7876b74a1af00eb73e25619c0036e0f82760df77 100644
--- a/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
+++ b/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
@@ -279,6 +279,7 @@ public class TitleScreen extends Screen {
             }
 
             guiGraphics.drawString(this.font, s, 2, this.height - 10, 16777215 | i3);
+            guiGraphics.drawString(this.font, "MagmaCube " + ((net.minecraft.DetectedVersion) SharedConstants.getCurrentVersion()).magmaCubeVersion, 2, this.height - 20, 0xFFFFFF | i3); // MagmaCube
 
             for(GuiEventListener guiEventListener : this.children()) {
                 if (guiEventListener instanceof AbstractWidget) {
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 803f81e3ecc96978caf9c82419bbcd5a04c53e9e..e7ca55d6860645896beffcd40a00eb23f455fa2f 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -949,7 +949,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
     @DontObfuscate
     public String getServerModName() {
-        return "vanilla";
+        return "magmacube"; // MagmaCube > vanilla!
     }
 
     public SystemReport fillSystemReport(SystemReport systemReport) {
