From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: acrylic-style <me@acrylicstyle.xyz>
Date: Mon, 11 Jan 2021 03:32:09 +0900
Subject: [PATCH] magmacube > vanilla!


diff --git a/src/main/java/net/minecraft/DetectedVersion.java b/src/main/java/net/minecraft/DetectedVersion.java
index c50996ee9dc0cf532c7b6f4e62c6d0a18a54727a..1b288d043121e30b90bf2b0fdd66f5155bfbe451 100644
--- a/src/main/java/net/minecraft/DetectedVersion.java
+++ b/src/main/java/net/minecraft/DetectedVersion.java
@@ -25,6 +25,7 @@ public class DetectedVersion implements WorldVersion {
     private final int resourcePackVersion;
     private final int dataPackVersion;
     private final Date buildTime;
+    @javax.annotation.Nullable public final String magmaCubeVersion; // MagmaCube
 
     private DetectedVersion() {
         this.id = UUID.randomUUID().toString().replaceAll("-", "");
@@ -35,6 +36,7 @@ public class DetectedVersion implements WorldVersion {
         this.resourcePackVersion = 12;
         this.dataPackVersion = 10;
         this.buildTime = new Date();
+        this.magmaCubeVersion = "debug"; // MagmaCube
     }
 
     private DetectedVersion(JsonObject json) {
@@ -47,6 +49,7 @@ public class DetectedVersion implements WorldVersion {
         this.resourcePackVersion = GsonHelper.getAsInt(jsonObject2, "resource");
         this.dataPackVersion = GsonHelper.getAsInt(jsonObject2, "data");
         this.buildTime = Date.from(ZonedDateTime.parse(GsonHelper.getAsString(json, "build_time")).toInstant());
+        this.magmaCubeVersion = json.has("magmacube_version") ? GsonHelper.getAsString(json, "magmacube_version") : "debug"; // MagmaCube
     }
 
     public static WorldVersion tryDetectVersion() {
diff --git a/src/main/java/net/minecraft/SystemReport.java b/src/main/java/net/minecraft/SystemReport.java
index a506f452e9357dd504701050482d3495e80fb8df..d407afcacaad89de4836abdc8a975f73570614ef 100644
--- a/src/main/java/net/minecraft/SystemReport.java
+++ b/src/main/java/net/minecraft/SystemReport.java
@@ -27,6 +27,7 @@ public class SystemReport {
     private final Map<String, String> entries = Maps.newLinkedHashMap();
 
     public SystemReport() {
+        this.setDetail("MagmaCube Version", ((DetectedVersion) SharedConstants.getCurrentVersion()).magmaCubeVersion); // MagmaCube
         this.setDetail("Minecraft Version", SharedConstants.getCurrentVersion().getName());
         this.setDetail("Minecraft Version ID", SharedConstants.getCurrentVersion().getId());
         this.setDetail("Operating System", OPERATING_SYSTEM);
diff --git a/src/main/java/net/minecraft/client/ClientBrandRetriever.java b/src/main/java/net/minecraft/client/ClientBrandRetriever.java
index a71fa408ba7e733c02f450dd5c18a17bee53a427..fb84bbcd532981037e97b33e7b8bdef539dbbd2c 100644
--- a/src/main/java/net/minecraft/client/ClientBrandRetriever.java
+++ b/src/main/java/net/minecraft/client/ClientBrandRetriever.java
@@ -3,10 +3,10 @@ package net.minecraft.client;
 import net.minecraft.obfuscate.DontObfuscate;
 
 public class ClientBrandRetriever {
-    public static final String VANILLA_NAME = "vanilla";
+    public static final String VANILLA_NAME = "magmacube"; // MagmaCube > vanilla!
 
     @DontObfuscate
     public static String getClientModName() {
-        return "vanilla";
+        return VANILLA_NAME;
     }
 }
\ No newline at end of file
diff --git a/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java b/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
index a7da012f117c34b505b4238f908f6d64c8b91a1e..cd03142651556e2c0c38f3405e541e8ffd8ba408 100644
--- a/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
+++ b/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
@@ -306,6 +306,7 @@ public class TitleScreen extends Screen {
                 s = s + I18n.get("menu.modded");
             }
 
+            drawString(poseStack, this.font, "MagmaCube " + ((net.minecraft.DetectedVersion) SharedConstants.getCurrentVersion()).magmaCubeVersion, 2, this.height - 20, 0xFFFFFF | i6); // MagmaCube
             drawString(poseStack, this.font, s, 2, this.height - 10, 16777215 | i6);
 
             for(GuiEventListener guiEventListener : this.children()) {
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 45364d9658a0c92d94fd784bd83639af3f705cd8..6907041caf3253dec59edac9b2dfba26127f9e33 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -925,7 +925,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
     @DontObfuscate
     public String getServerModName() {
-        return "vanilla";
+        return "magmacube"; // MagmaCube > vanilla!
     }
 
     public SystemReport fillSystemReport(SystemReport systemReport) {
