From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: acrylic-style <me@acrylicstyle.xyz>
Date: Mon, 11 Jan 2021 03:32:09 +0900
Subject: [PATCH] magmacube > vanilla!


diff --git a/src/main/java/net/minecraft/DetectedVersion.java b/src/main/java/net/minecraft/DetectedVersion.java
index 33596cc4f0edaa27d69261071c1db9b90510a5cb..679c78c0978cf0f7f179cb307e9534e7d099f92a 100644
--- a/src/main/java/net/minecraft/DetectedVersion.java
+++ b/src/main/java/net/minecraft/DetectedVersion.java
@@ -26,6 +26,7 @@ public class DetectedVersion implements WorldVersion {
     private final int dataPackVersion;
     private final Date buildTime;
     private final String releaseTarget;
+    @javax.annotation.Nullable public final String magmaCubeVersion; // MagmaCube
 
     private DetectedVersion() {
         this.id = UUID.randomUUID().toString().replaceAll("-", "");
@@ -37,6 +38,7 @@ public class DetectedVersion implements WorldVersion {
         this.dataPackVersion = 10;
         this.buildTime = new Date();
         this.releaseTarget = "1.19.1";
+        this.magmaCubeVersion = "debug"; // MagmaCube
     }
 
     private DetectedVersion(JsonObject json) {
@@ -50,6 +52,7 @@ public class DetectedVersion implements WorldVersion {
         this.resourcePackVersion = GsonHelper.getAsInt(jsonObject2, "resource");
         this.dataPackVersion = GsonHelper.getAsInt(jsonObject2, "data");
         this.buildTime = Date.from(ZonedDateTime.parse(GsonHelper.getAsString(json, "build_time")).toInstant());
+        this.magmaCubeVersion = json.has("magmacube_version") ? GsonHelper.getAsString(json, "magmacube_version") : "debug"; // MagmaCube
     }
 
     public static WorldVersion tryDetectVersion() {
diff --git a/src/main/java/net/minecraft/SystemReport.java b/src/main/java/net/minecraft/SystemReport.java
index 04e68a7fed2381d420567c4b52577c94cb43cc2a..da5ac7b22dd629ec1fffcdabc35743c161c5d611 100644
--- a/src/main/java/net/minecraft/SystemReport.java
+++ b/src/main/java/net/minecraft/SystemReport.java
@@ -26,6 +26,7 @@ public class SystemReport {
     private final Map<String, String> entries = Maps.newLinkedHashMap();
 
     public SystemReport() {
+        this.setDetail("MagmaCube Version", ((DetectedVersion) SharedConstants.getCurrentVersion()).magmaCubeVersion); // MagmaCube
         this.setDetail("Minecraft Version", SharedConstants.getCurrentVersion().getName());
         this.setDetail("Minecraft Version ID", SharedConstants.getCurrentVersion().getId());
         this.setDetail("Operating System", OPERATING_SYSTEM);
diff --git a/src/main/java/net/minecraft/client/ClientBrandRetriever.java b/src/main/java/net/minecraft/client/ClientBrandRetriever.java
index a71fa408ba7e733c02f450dd5c18a17bee53a427..fb84bbcd532981037e97b33e7b8bdef539dbbd2c 100644
--- a/src/main/java/net/minecraft/client/ClientBrandRetriever.java
+++ b/src/main/java/net/minecraft/client/ClientBrandRetriever.java
@@ -3,10 +3,10 @@ package net.minecraft.client;
 import net.minecraft.obfuscate.DontObfuscate;
 
 public class ClientBrandRetriever {
-    public static final String VANILLA_NAME = "vanilla";
+    public static final String VANILLA_NAME = "magmacube"; // MagmaCube > vanilla!
 
     @DontObfuscate
     public static String getClientModName() {
-        return "vanilla";
+        return VANILLA_NAME;
     }
 }
\ No newline at end of file
diff --git a/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java b/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
index 03273b019b87f27345e8ddaa92ec558e6c42439f..91c4d21d32e80cb7c0d686a0bbd915cc3972e1cb 100644
--- a/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
+++ b/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
@@ -314,6 +314,7 @@ public class TitleScreen extends Screen {
                 s = s + I18n.get("menu.modded");
             }
 
+            drawString(poseStack, this.font, "MagmaCube " + ((net.minecraft.DetectedVersion) SharedConstants.getCurrentVersion()).magmaCubeVersion, 2, this.height - 20, 0xFFFFFF | i6); // MagmaCube
             drawString(poseStack, this.font, s, 2, this.height - 10, 16777215 | i6);
 
             for(GuiEventListener guiEventListener : this.children()) {
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index c1c40ca0fea4d891274662d1fde173efb7433323..40971e87b03fd4689bd41e3bebf03e76ab8ae43e 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -918,7 +918,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
     @DontObfuscate
     public String getServerModName() {
-        return "vanilla";
+        return "magmacube"; // MagmaCube > vanilla!
     }
 
     public SystemReport fillSystemReport(SystemReport systemReport) {
