From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: acrylic-style <me@acrylicstyle.xyz>
Date: Mon, 11 Jan 2021 03:55:35 +0900
Subject: [PATCH] Fix incorrect byte order


diff --git a/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java b/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java
index aec550052d87b4efd229723a3f8defa3c1b6b4d5..49d3135513890f0d54cec50dc74aeccb0c893b19 100644
--- a/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java
+++ b/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java
@@ -321,6 +321,7 @@ public class BufferBuilder extends DefaultedVertexConsumer implements BufferVert
         }
 
         ByteBuffer byteBuffer = this.buffer.slice();
+        byteBuffer.order(this.buffer.order()); // MagmaCube - fix incorrect byte order
         this.buffer.clear();
         return Pair.of(drawState, byteBuffer);
     }
