From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: acrylic-style <ilyy@outlook.jp>
Date: Mon, 11 Jan 2021 03:32:09 +0900
Subject: [PATCH] magmacube > vanilla!


diff --git a/src/main/java/net/minecraft/DetectedVersion.java b/src/main/java/net/minecraft/DetectedVersion.java
index b7bd976876a29d277df2e9eaa488d206662fce20..9a13c492a03c4b9ca4d7db561acc47c9e3769f01 100644
--- a/src/main/java/net/minecraft/DetectedVersion.java
+++ b/src/main/java/net/minecraft/DetectedVersion.java
@@ -26,6 +26,7 @@ public class DetectedVersion implements GameVersion {
    private final int dataPackVersion;
    private final Date buildTime;
    private final String releaseTarget;
+   @javax.annotation.Nullable public final String magmaCubeVersion; // MagmaCube
 
    private DetectedVersion() {
       this.id = UUID.randomUUID().toString().replaceAll("-", "");
@@ -37,6 +38,7 @@ public class DetectedVersion implements GameVersion {
       this.dataPackVersion = 7;
       this.buildTime = new Date();
       this.releaseTarget = "1.17.1";
+      this.magmaCubeVersion = "debug"; // MagmaCube
    }
 
    private DetectedVersion(JsonObject json) {
@@ -50,6 +52,7 @@ public class DetectedVersion implements GameVersion {
       this.resourcePackVersion = GsonHelper.getAsInt(jsonObject2, "resource");
       this.dataPackVersion = GsonHelper.getAsInt(jsonObject2, "data");
       this.buildTime = Date.from(ZonedDateTime.parse(GsonHelper.getAsString(json, "build_time")).toInstant());
+      this.magmaCubeVersion = json.has("magmacube_version") ? GsonHelper.getAsString(json, "magmacube_version") : "debug"; // MagmaCube
    }
 
    public static GameVersion tryDetectVersion() {
diff --git a/src/main/java/net/minecraft/SystemReport.java b/src/main/java/net/minecraft/SystemReport.java
index 54acc476129cc4690cae1c643bc90022b5d21e5d..7de71eb2eede551afce27d6c90a31e277be09419 100644
--- a/src/main/java/net/minecraft/SystemReport.java
+++ b/src/main/java/net/minecraft/SystemReport.java
@@ -26,6 +26,7 @@ public class SystemReport {
    private final Map<String, String> entries = Maps.newLinkedHashMap();
 
    public SystemReport() {
+      this.setDetail("MagmaCube Version", ((DetectedVersion) SharedConstants.getCurrentVersion()).magmaCubeVersion); // MagmaCube
       this.setDetail("Minecraft Version", SharedConstants.getCurrentVersion().getName());
       this.setDetail("Minecraft Version ID", SharedConstants.getCurrentVersion().getId());
       this.setDetail("Operating System", OPERATING_SYSTEM);
diff --git a/src/main/java/net/minecraft/client/ClientBrandRetriever.java b/src/main/java/net/minecraft/client/ClientBrandRetriever.java
index 1163b0e7de403ce1782587f48e42c37be0494e45..8c3e951c3f6a7f6a4e8e150cb325847303ded360 100644
--- a/src/main/java/net/minecraft/client/ClientBrandRetriever.java
+++ b/src/main/java/net/minecraft/client/ClientBrandRetriever.java
@@ -3,10 +3,10 @@ package net.minecraft.client;
 import net.minecraft.obfuscate.DontObfuscate;
 
 public class ClientBrandRetriever {
-   public static final String VANILLA_NAME = "vanilla";
+   public static final String VANILLA_NAME = "magmacube"; // MagmaCube > vanilla!;
 
    @DontObfuscate
    public static String getClientModName() {
-      return "vanilla";
+      return VANILLA_NAME;
    }
 }
\ No newline at end of file
diff --git a/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java b/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
index f2a7d33c5f3d2a570b6399424e3d3a8ef4c3d730..84e3d01e69d2bd65706ef763cf6cc821f92115cc 100644
--- a/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
+++ b/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
@@ -99,7 +99,7 @@ public class TitleScreen extends Screen {
          this.splash = this.minecraft.getSplashManager().getSplash();
       }
 
-      this.copyrightWidth = this.font.width("Copyright Mojang AB. Do not distribute!");
+      this.copyrightWidth = this.font.width(COPYRIGHT_TEXT);
       this.copyrightX = this.width - this.copyrightWidth - 2;
       int i = 24;
       int i2 = this.height / 4 + 48;
@@ -290,8 +290,9 @@ public class TitleScreen extends Screen {
             s = s + I18n.get("menu.modded");
          }
 
+         drawString(poseStack, this.font, "MagmaCube " + ((net.minecraft.DetectedVersion) SharedConstants.getCurrentVersion()).magmaCubeVersion, 2, this.height - 20, 0xFFFFFF | i6); // MagmaCube
          drawString(poseStack, this.font, s, 2, this.height - 10, 16777215 | i6);
-         drawString(poseStack, this.font, "Copyright Mojang AB. Do not distribute!", this.copyrightX, this.height - 10, 16777215 | i6);
+         drawString(poseStack, this.font, COPYRIGHT_TEXT, this.copyrightX, this.height - 10, 16777215 | i6);
          if (i > this.copyrightX && i < this.copyrightX + this.copyrightWidth && i2 > this.height - 10 && i2 < this.height) {
             fill(poseStack, this.copyrightX, this.height - 1, this.copyrightX + this.copyrightWidth, this.height, 16777215 | i6);
          }
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 21db712c7bedf63551e70e11c86b81a4535962ce..2f2adb2fc7df4150bb6eaecd4254fcdad6f08782 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -936,7 +936,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
    @DontObfuscate
    public String getServerModName() {
-      return "vanilla";
+      return "magmacube"; // MagmaCube > vanilla!
    }
 
    public SystemReport fillSystemReport(SystemReport systemReport) {
