From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: acrylic-style <ilyy@outlook.jp>
Date: Mon, 11 Jan 2021 03:32:09 +0900
Subject: [PATCH] magmacube > vanilla!


diff --git a/src/main/java/net/minecraft/CrashReport.java b/src/main/java/net/minecraft/CrashReport.java
index 67b27413f6167ffa0cd2d530c954aeb55b4cba3a..f5432c828930a23fbd5657a978c1c5db39a0a4ca 100644
--- a/src/main/java/net/minecraft/CrashReport.java
+++ b/src/main/java/net/minecraft/CrashReport.java
@@ -35,6 +35,7 @@ public class CrashReport {
    }
 
    private void initDetails() {
+      this.systemDetails.setDetail("MagmaCube Version", () -> ((DetectedVersion) SharedConstants.getCurrentVersion()).magmaCubeVersion); // MagmaCube
       this.systemDetails.setDetail("Minecraft Version", (CrashReportDetail<String>)(() -> SharedConstants.getCurrentVersion().getName()));
       this.systemDetails.setDetail("Minecraft Version ID", (CrashReportDetail<String>)(() -> SharedConstants.getCurrentVersion().getId()));
       this.systemDetails.setDetail("Operating System", (CrashReportDetail<String>)(() -> System.getProperty("os.name") + " (" + System.getProperty("os.arch") + ") version " + System.getProperty("os.version")));
diff --git a/src/main/java/net/minecraft/DetectedVersion.java b/src/main/java/net/minecraft/DetectedVersion.java
index 2636e5f8bca8f677970c14baa46cc67970c5c07f..8a5bf16e50979c59ac09999d1d232af1745c33bf 100644
--- a/src/main/java/net/minecraft/DetectedVersion.java
+++ b/src/main/java/net/minecraft/DetectedVersion.java
@@ -26,6 +26,7 @@ public class DetectedVersion implements GameVersion {
    private final int dataPackVersion;
    private final Date buildTime;
    private final String releaseTarget;
+   @javax.annotation.Nullable public final String magmaCubeVersion; // MagmaCube
 
    private DetectedVersion() {
       this.id = UUID.randomUUID().toString().replaceAll("-", "");
@@ -37,6 +38,7 @@ public class DetectedVersion implements GameVersion {
       this.dataPackVersion = 7;
       this.buildTime = new Date();
       this.releaseTarget = "1.17";
+      this.magmaCubeVersion = "debug"; // MagmaCube
    }
 
    private DetectedVersion(JsonObject json) {
@@ -50,6 +52,7 @@ public class DetectedVersion implements GameVersion {
       this.resourcePackVersion = GsonHelper.getAsInt(jsonObject2, "resource");
       this.dataPackVersion = GsonHelper.getAsInt(jsonObject2, "data");
       this.buildTime = Date.from(ZonedDateTime.parse(GsonHelper.getAsString(json, "build_time")).toInstant());
+      this.magmaCubeVersion = json.has("magmacube_version") ? GsonHelper.getAsString(json, "magmacube_version") : "debug";
    }
 
    public static GameVersion tryDetectVersion() {
diff --git a/src/main/java/net/minecraft/client/ClientBrandRetriever.java b/src/main/java/net/minecraft/client/ClientBrandRetriever.java
index 1b2cf23c6f943151d84093a3f837d36cccbeb493..4d80e4054f8436bd8fd1af85a8121a70304a256f 100644
--- a/src/main/java/net/minecraft/client/ClientBrandRetriever.java
+++ b/src/main/java/net/minecraft/client/ClientBrandRetriever.java
@@ -2,6 +2,6 @@ package net.minecraft.client;
 
 public class ClientBrandRetriever {
    public static String getClientModName() {
-      return "vanilla";
+      return "magmacube"; // MagmaCube > vanilla!
    }
 }
\ No newline at end of file
diff --git a/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java b/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
index 1ff81105f20e2ee1198567b614f5b24c601c246f..fc9ddc6ef836797a921822c2e61e37c4407e3ea5 100644
--- a/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
+++ b/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
@@ -238,6 +238,7 @@ public class TitleScreen extends Screen {
             s = s + I18n.get("menu.modded");
          }
 
+         drawString(poseStack, this.font, "MagmaCube " + ((net.minecraft.DetectedVersion) SharedConstants.getCurrentVersion()).magmaCubeVersion, 2, this.height - 20, 0xFFFFFF | i6); // MagmaCube
          drawString(poseStack, this.font, s, 2, this.height - 10, 16777215 | i6);
          drawString(poseStack, this.font, "Copyright Mojang AB. Do not distribute!", this.copyrightX, this.height - 10, 16777215 | i6);
          if (i > this.copyrightX && i < this.copyrightX + this.copyrightWidth && i2 > this.height - 10 && i2 < this.height) {
