From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: acrylic-style <me@acrylicstyle.xyz>
Date: Mon, 11 Jan 2021 03:32:09 +0900
Subject: [PATCH] magmacube > vanilla!


diff --git a/src/main/java/net/minecraft/DetectedVersion.java b/src/main/java/net/minecraft/DetectedVersion.java
index 374fecc1fec7fb31bd0b4eda2ef1de6fcce45027..91abf064c0c3a7dae6435dd922dc3a4d29b47ab3 100644
--- a/src/main/java/net/minecraft/DetectedVersion.java
+++ b/src/main/java/net/minecraft/DetectedVersion.java
@@ -26,6 +26,7 @@ public class DetectedVersion implements WorldVersion {
     private final int dataPackVersion;
     private final Date buildTime;
     private final String releaseTarget;
+    @javax.annotation.Nullable public final String magmaCubeVersion; // MagmaCube
 
     private DetectedVersion() {
         this.id = UUID.randomUUID().toString().replaceAll("-", "");
@@ -37,6 +38,7 @@ public class DetectedVersion implements WorldVersion {
         this.dataPackVersion = 8;
         this.buildTime = new Date();
         this.releaseTarget = "1.18.1";
+        this.magmaCubeVersion = "debug"; // MagmaCube
     }
 
     private DetectedVersion(JsonObject json) {
@@ -50,6 +52,7 @@ public class DetectedVersion implements WorldVersion {
         this.resourcePackVersion = GsonHelper.getAsInt(jsonObject2, "resource");
         this.dataPackVersion = GsonHelper.getAsInt(jsonObject2, "data");
         this.buildTime = Date.from(ZonedDateTime.parse(GsonHelper.getAsString(json, "build_time")).toInstant());
+        this.magmaCubeVersion = json.has("magmacube_version") ? GsonHelper.getAsString(json, "magmacube_version") : "debug"; // MagmaCube
     }
 
     public static WorldVersion tryDetectVersion() {
diff --git a/src/main/java/net/minecraft/SystemReport.java b/src/main/java/net/minecraft/SystemReport.java
index 77f13c0c2ec82fbf82e400a035bbb249fe359afb..74e8f78d68f387fc5d99e42999b7a63a8103f77c 100644
--- a/src/main/java/net/minecraft/SystemReport.java
+++ b/src/main/java/net/minecraft/SystemReport.java
@@ -26,6 +26,7 @@ public class SystemReport {
     private final Map<String, String> entries = Maps.newLinkedHashMap();
 
     public SystemReport() {
+        this.setDetail("MagmaCube Version", ((DetectedVersion) SharedConstants.getCurrentVersion()).magmaCubeVersion); // MagmaCube
         this.setDetail("Minecraft Version", SharedConstants.getCurrentVersion().getName());
         this.setDetail("Minecraft Version ID", SharedConstants.getCurrentVersion().getId());
         this.setDetail("Operating System", OPERATING_SYSTEM);
diff --git a/src/main/java/net/minecraft/client/ClientBrandRetriever.java b/src/main/java/net/minecraft/client/ClientBrandRetriever.java
index a71fa408ba7e733c02f450dd5c18a17bee53a427..fb84bbcd532981037e97b33e7b8bdef539dbbd2c 100644
--- a/src/main/java/net/minecraft/client/ClientBrandRetriever.java
+++ b/src/main/java/net/minecraft/client/ClientBrandRetriever.java
@@ -3,10 +3,10 @@ package net.minecraft.client;
 import net.minecraft.obfuscate.DontObfuscate;
 
 public class ClientBrandRetriever {
-    public static final String VANILLA_NAME = "vanilla";
+    public static final String VANILLA_NAME = "magmacube"; // MagmaCube > vanilla!
 
     @DontObfuscate
     public static String getClientModName() {
-        return "vanilla";
+        return VANILLA_NAME;
     }
 }
\ No newline at end of file
diff --git a/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java b/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
index 40d4d30f75d514c00e1cc6a252faae3fea503058..4b8d68438c9ead9ac3306a75ffa19e478882d323 100644
--- a/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
+++ b/src/main/java/net/minecraft/client/gui/screens/TitleScreen.java
@@ -100,7 +100,7 @@ public class TitleScreen extends Screen {
             this.splash = this.minecraft.getSplashManager().getSplash();
         }
 
-        this.copyrightWidth = this.font.width("Copyright Mojang AB. Do not distribute!");
+        this.copyrightWidth = this.font.width(COPYRIGHT_TEXT);
         this.copyrightX = this.width - this.copyrightWidth - 2;
         int i = 24;
         int i2 = this.height / 4 + 48;
@@ -291,8 +291,9 @@ public class TitleScreen extends Screen {
                 s = s + I18n.get("menu.modded");
             }
 
+            drawString(poseStack, this.font, "MagmaCube " + ((net.minecraft.DetectedVersion) SharedConstants.getCurrentVersion()).magmaCubeVersion, 2, this.height - 20, 0xFFFFFF | i6); // MagmaCube
             drawString(poseStack, this.font, s, 2, this.height - 10, 16777215 | i6);
-            drawString(poseStack, this.font, "Copyright Mojang AB. Do not distribute!", this.copyrightX, this.height - 10, 16777215 | i6);
+            drawString(poseStack, this.font, COPYRIGHT_TEXT, this.copyrightX, this.height - 10, 16777215 | i6);
             if (i > this.copyrightX && i < this.copyrightX + this.copyrightWidth && i2 > this.height - 10 && i2 < this.height) {
                 fill(poseStack, this.copyrightX, this.height - 1, this.copyrightX + this.copyrightWidth, this.height, 16777215 | i6);
             }
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 60be6da816a824e06a4370e5f1571cbb40f299af..2495572ee0b0dce2beb87d1c612ad2965e1af99c 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -922,7 +922,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
     @DontObfuscate
     public String getServerModName() {
-        return "vanilla";
+        return "magmacube"; // MagmaCube > vanilla!
     }
 
     public SystemReport fillSystemReport(SystemReport systemReport) {
