From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: acrylic-style <ilyy@outlook.jp>
Date: Mon, 11 Jan 2021 03:55:35 +0900
Subject: [PATCH] Fix incorrect byte order


diff --git a/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java b/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java
index 4bf874107f3317ff8de9aa798497d269134d664e..82ad895d7baee3e852a51ab76f24ee5c68af0b25 100644
--- a/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java
+++ b/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java
@@ -323,6 +323,7 @@ public class BufferBuilder extends DefaultedVertexConsumer implements BufferVert
       }
 
       ByteBuffer byteBuffer = this.buffer.slice();
+      byteBuffer.order(this.buffer.order()); // MagmaCube - fix incorrect byte order
       this.buffer.clear();
       return Pair.of(drawState, byteBuffer);
    }
