From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: acrylic-style <me@acrylicstyle.xyz>
Date: Mon, 11 Jan 2021 03:55:35 +0900
Subject: [PATCH] Fix incorrect byte order


diff --git a/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java b/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java
index 3d487f375e5dc3e4deb276ad8a7164e65848c03a..c280ac900e00143072c6c23bf118709ae174c99e 100644
--- a/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java
+++ b/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java
@@ -321,6 +321,7 @@ public class BufferBuilder extends DefaultedVertexConsumer implements BufferVert
         }
 
         ByteBuffer byteBuffer = this.buffer.slice();
+        byteBuffer.order(this.buffer.order()); // MagmaCube - fix incorrect byte order
         this.buffer.clear();
         return Pair.of(drawState, byteBuffer);
     }
