From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: acrylic-style <ilyy@outlook.jp>
Date: Mon, 11 Jan 2021 03:55:35 +0900
Subject: [PATCH] Fix incorrect byte order


diff --git a/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java b/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java
index 66b08b0db293b5a408e8351a0e27d4594d6a7537..8b5f063e27537ef7ad88d1c4e072a3874eb03ef9 100644
--- a/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java
+++ b/src/main/java/com/mojang/blaze3d/vertex/BufferBuilder.java
@@ -321,6 +321,7 @@ public class BufferBuilder extends DefaultedVertexConsumer implements BufferVert
       }
 
       ByteBuffer byteBuffer = this.buffer.slice();
+      byteBuffer.order(this.buffer.order()); // MagmaCube - fix incorrect byte order
       this.buffer.clear();
       return Pair.of(drawState, byteBuffer);
    }
