From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: acrylic-style <ilyy@outlook.jp>
Date: Tue, 12 Jan 2021 22:09:02 +0900
Subject: [PATCH] Add ability to ignore packet error


diff --git a/src/main/java/net/minecraft/network/PacketDecoder.java b/src/main/java/net/minecraft/network/PacketDecoder.java
index 7bffd4c7d7caf182de6d8b7ce1ed42562c364948..942624cf8dbe5f3a7a1441589be3b45f19cc8161 100644
--- a/src/main/java/net/minecraft/network/PacketDecoder.java
+++ b/src/main/java/net/minecraft/network/PacketDecoder.java
@@ -17,6 +17,7 @@ public class PacketDecoder extends ByteToMessageDecoder {
    private static final Logger LOGGER = LogManager.getLogger();
    private static final Marker MARKER = MarkerManager.getMarker("PACKET_RECEIVED", Connection.PACKET_MARKER);
    private final PacketFlow flow;
+   private final boolean ignoreError = Boolean.parseBoolean(System.getProperty("magmacube.ignorePacketError", System.getenv("magmacubeIgnorePacketError"))); // MagmaCube
 
    public PacketDecoder(PacketFlow packetFlow) {
       this.flow = packetFlow;
@@ -39,7 +40,13 @@ public class PacketDecoder extends ByteToMessageDecoder {
             }
 
             if (friendlyByteBuf.readableBytes() > 0) {
-               throw new IOException("Packet " + ((ConnectionProtocol)channelHandlerContext.channel().attr(Connection.ATTRIBUTE_PROTOCOL).get()).getId() + "/" + i2 + " (" + packet.getClass().getSimpleName() + ") was larger than I expected, found " + friendlyByteBuf.readableBytes() + " bytes extra whilst reading packet " + i2);
+               // MagmaCube start - add ability to ignore packet error
+               if (ignoreError) {
+                  LOGGER.warn("Packet " + ((ConnectionProtocol) channelHandlerContext.channel().attr(Connection.ATTRIBUTE_PROTOCOL).get()).getId() + "/" + i + " (" + packet.getClass().getSimpleName() + ") was larger than I expected, found " + friendlyByteBuf.readableBytes() + " bytes extra whilst reading packet " + i2);
+                  return;
+               }
+               // MagmaCube end - add ability to ignore packet error
+               throw new IOException("Packet " + ((ConnectionProtocol)channelHandlerContext.channel().attr(Connection.ATTRIBUTE_PROTOCOL).get()).getId() + "/" + i + " (" + packet.getClass().getSimpleName() + ") was larger than I expected, found " + friendlyByteBuf.readableBytes() + " bytes extra whilst reading packet " + i2);
             } else {
                list.add(packet);
                if (LOGGER.isDebugEnabled()) {
